mod/consensus/pkg/cometbft/service/validate_execution_header.go

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: BUSL-1.1
+//
+// Copyright (C) 2024, Berachain Foundation. All rights reserved.
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file of this repository and at www.mariadb.com/bsl11.
+//
+// ANY USE OF THE LICENSED WORK IN VIOLATION OF THIS LICENSE WILL AUTOMATICALLY
+// TERMINATE YOUR RIGHTS UNDER THIS LICENSE FOR THE CURRENT AND ALL OTHER
+// VERSIONS OF THE LICENSED WORK.
+//
+// THIS LICENSE DOES NOT GRANT YOU ANY RIGHT IN ANY TRADEMARK OR LOGO OF
+// LICENSOR OR ITS AFFILIATES (PROVIDED THAT YOU MAY USE A TRADEMARK OR LOGO OF
+// LICENSOR AS EXPRESSLY REQUIRED BY THIS LICENSE).
+//
+// TO THE EXTENT PERMITTED BY APPLICABLE LAW, THE LICENSED WORK IS PROVIDED ON
+// AN “AS IS” BASIS. LICENSOR HEREBY DISCLAIMS ALL WARRANTIES AND CONDITIONS,
+// EXPRESS OR IMPLIED, INCLUDING (WITHOUT LIMITATION) WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, NON-INFRINGEMENT, AND
+// TITLE.
+
+package cometbft
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/berachain/beacon-kit/mod/consensus-types/pkg/types"
+	"github.com/berachain/beacon-kit/mod/errors"
+	"github.com/berachain/beacon-kit/mod/primitives/pkg/common"
+)
+
+// maxExtraDataSize defines the maximum allowed size in bytes for the ExtraData
+// field in the execution payload header.
+const maxExtraDataSize = 32
+
+// validateExecutionHeader validates the provided execution payload header
+// for the genesis block.
+func validateExecutionHeader(header *types.ExecutionPayloadHeader) error {
```

Comment:
I checked the genesis data spinning up a network locally via `make start` and `make start-geth`- TransactionsRoot and WithdrawalsRoot are not empty and same goes for stateRoot.
This is how my emptyTrie looks like 
`	emptyTrieRoot := common.Bytes32(
		common.NewExecutionHashFromHex(
			"0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
	))` 
which is not matching with `stateRoot`. Do you know why it could be so ? @calbera

Reply:
Removed the newly introduced validation for empty roots as it was failing and also removed validation for zero, seems they are not really required.